testdata/project-v2/PROJECT

```diff
@@ -1,13 +1,13 @@
-version: "2"
 domain: testproject.org
 repo: sigs.k8s.io/kubebuilder/testdata/project-v2
 resources:
 - group: crew
-  version: v1
   kind: Captain
-- group: crew
   version: v1
-  kind: FirstMate
 - group: crew
+  kind: FirstMate
   version: v1
+- group: crew
   kind: Admiral
+  version: v1
+version: "2"
```

Comment:
Can we control the order somehow? It is using alphabetical ordering, and while this may not be improtant for resources, having `version: "X"` before the resources list seems pretty convenient for both users and scaffolding extra resources as we will just be appending to the file instead of inserting in the middle.

Reply:
It is not inserted in the middle and it is generated automatically. In this way, any change regards to it will not be part of the scope of this PR.

Reply:
I know this file is generated automatically and that the scaffold is able to insert in the middle. I just asked if the current behavior can be achieved as this may lead to future optimizations (appending to a file requires less time from the underlying os than creating it from scratch most of the times). If it can not be achieved, I think it will be ok, but if it can be achieved it may be interesting to keep the order as is.

Reply:
Hi @Adirio

Thank you for your input.  However, IHMO too many comments or too much verbose can make harder the pr get reviewed and merged. So, any subject which is not revalent to the pr scope should not be addressed here. 

But note that you are able always to address all your ideas/suggestions by raising new issues which I think would be more appropriate in this kind of scenario. WDYT?

Reply:
I do not agree that this comment is out of scope of this PR. This PR is changing the YAML parser, which results in a different field order. As this PR is changing current behavior (version is at the top and resources at the end) asking if the order can be kept the same its IMHO in the scope of this project.
About the length of comments, thats what the "mark conversation as resolved" is for. Once the author solves the comment (in this case that means either keeping the current order or saying it cannot be done but the owners of the repo still find it ok) this comment can be marked as solved and it won't take any space.
Offtopi: what it is adding noise is all this conversation about the ammount of comments

Reply:
Thanks for the comment, I don't believe this is possible at least not to reflect it's original output, albeit it seems like it might be ordering in alphabetical order which is at least interesting. Resolving.

Reply:
I'm OK as long as the order is deterministic and stable.